### PR TITLE
refactor openlibrary/core/lending.py to use requests

### LINKED ISSUE #2852: Replace all urllib / urllib2 imports and uses with `requests`
`urllib2` is old, and a bit clunky -- I've found recently it can't do some of things I needed it to do, and it also causes problem for moving to Python3, and generally seems to make the code more complex.

Instead of re-using it, and adding new features that use the older mechanisms, can we consider refactoring to use `requests` instead?

My PR where I stripped out `urllib2` and `StringIO` etc to enable `HEAD` requests:  #2838

a similar PR that went with using `requests` to avoid old bugs: #2779 

By using `requests` (rather than adding imports from `six`) we should be getting a better tool for making requests, and reducing code complexity, and the number of imports. 

Relates to #2308 since a common use of `simplejson` is to parse `urllib2` responses. `requests` has a builtin `json()` method so an extra import is not required.

### PR DESCRIPTION
**Pull request by aasifkhan7**
Related to #2852 

### Stakeholders

@cclauss @mekarpeles

### CODE REVIEW

**Comment by aasifkhan7** [comment]
Related to https://github.com/internetarchive/openlibrary/pull/3658

**Review by cclauss — APPROVED** [review]
LGTM.  Thanks!

File: openlibrary/core/lending.py [hunk 1]
```diff
@@ -213,8 +211,8 @@ def get_random_available_ia_edition():
                "+AND+loans__status__status:AVAILABLE"
                "&fl=identifier,openlibrary_edition,loans__status__status"
                "&output=json&rows=1&sort[]=random" % (config_bookreader_host))
-        content = urllib.request.urlopen(url=url, timeout=config_http_request_timeout).read()
-        items = simplejson.loads(content).get('response', {}).get('docs', [])
+        response = requests.get(url, timeout=config_http_request_timeout)
+        items = response.json().get('response', {}).get('docs', [])
```
Comment (cclauss) [inline openlibrary/core/lending.py:215]: https://staging.openlibrary.org/random worked as expected and results were always available to borrow.

File: openlibrary/core/lending.py [hunk 2]
```diff
@@ -278,16 +276,15 @@ def get_available(limit=None, page=1, subject=None, query=None,
...
         # we can attribute requests to end-users
         client_ip = web.ctx.env.get('HTTP_X_FORWARDED_FOR', 'ol-internal')
-        request.add_header('x-client-id', client_ip)
 
-        content = urllib.request.urlopen(request, timeout=config_http_request_timeout).read()
-        items = simplejson.loads(content).get('response', {}).get('docs', [])
+        response = requests.get(
+            url, headers={"x-client-id": client_ip}, timeout=config_http_request_timeout
+        )
+        items = response.json().get('response', {}).get('docs', [])
```
Comment (cclauss) [inline openlibrary/core/lending.py:287]: http://staging.openlibrary.org/books/OL13439647M/The_brothers_Karamazov did not raise an exception as it did in (https://sentry.archive.org/sentry/ol-web/issues/3873) on Python 2 but it was not available for borrow.

**Pull request merged**